docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,101 @@
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image for append.
+
+- Image types: Full ISO, minimal ISO, iPXE
+- Deployment types: Cloud, operator
+
+API which supports full set of operations as defined by Coreos Installer: append, replace, delete.
+
+### Non-Goals
+Manipulation of existing kernel parameters (replace/delete).  Replace and delete of exiting
+kernel arguments in ISO file is supported by Coreos Installer, but this implementation will not 
+support it. 
+
+## Proposal
+
+The discovery kernel arguments will be stored in the infra-env database record in a designated JSON encoded field.
+During infra-env registration and update the full set of the kernel arguments can be set.
+
+Setting the arguments can either be done by either REST API or Kube API (Part of InfraEnv CR)
+### User Stories
+
+#### Story 1
+
+When having slow NIC, increase the timeout to have the IP address configured.
+
+### Implementation Details/Notes/Constraints
+
+#### iPXE implementation
+
+iPXE implementation should be quite straight forward.  The arguments will be concatenated at the end of the 
+**kernel** command which is part of the iPXE script.
+
+#### Minimal and full ISO
+
+When streaming the discovery ISO file, the kernel arguments are included
+as part of some files: grub.cfg and isolinux.cfg
+
+Each of these files contains an area where the kernel arguments can be
+embedded. This can be identified by the regex **"(\n#+) COREOS_KARG_EMBED_AREA"**.
+The arguments should overwrite this are from the beginning and should
+not pass the last # (hash sign). The added kernel will be injected in the beginning.
+These arguments will be seperated by space **(" ")**.  When embedding the kernel arguments,
+the size of the above files will not change.
+
+In new Openshift versions, there is a file that contains the file names,
+while in old versions the file does not exist. If the file exists, the
+files to embed the kernel arguments are taken from this config file.
+Otherwise, the default files **[/EFI/redhat/grub.cfg, /isolinux/isolinux.cfg]**
+are used.
+
+
+We will need the image service to query the assisted service to get the kernel arguments.
+This will require authentication.  The authentication methodology may have 2 variants:  
+
+1. When running in cloud
+2. When running in disconnected environment
+
+The image service already queries the assisted service.  We will probably use the same authentication 
+methods.  We will need to verify that the infra-env endpoint may be accessed using the same authentication
+mechanism.
```

Comment:
I think we're sure of all of this now, right?

You can drop "probably" and state directly what we're doing.
If there's more to investigate I think this should be moved to "Open Questions"

Reply:
OK